src/components/znkExercise/core/directive/znkExerciseBtnSectionDrv.js

```diff
@@ -50,14 +50,15 @@
                         }
 
                         function _isLastQuestion(index, questions) {
-                            return (index && index === (questions.length - 1) ) || znkExerciseDrvCtrl.isLastUnansweredQuestion();
+                            return index && index === (questions.length - 1);
                         }
 
-                        function _setDoneBtnDisplayStatus(currIndex){
+                        function _setDoneBtnDisplayStatus(currIndex) {
                             var getQuestionsProm = znkExerciseDrvCtrl.getQuestions();
+                            var answeredCount = znkExerciseDrvCtrl.answeredCount;
                             getQuestionsProm.then(function (questions) {
                                 scope.vm.maxQuestionIndex = questions.length - 1;
-                                if (_notReviewMode() && _isLastQuestion(currIndex, questions)) {
+                                if (_notReviewMode() && (_isLastQuestion(currIndex, questions) || answeredCount === questions.length)) {
```

Comment:
Please add it as a function to znk exercise controller, areAllQuestionsAnswered.